app/gosns/publish_batch.go

```diff
@@ -0,0 +1,111 @@
+package gosns
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/common"
+	"github.com/Admiral-Piett/goaws/app/interfaces"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/utils"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func PublishBatchV1(req *http.Request) (int, interfaces.AbstractResponseBody) {
+	requestBody := models.NewPublishBatchRequest()
+	ok := utils.REQUEST_TRANSFORMER(requestBody, req, false)
+	if !ok {
+		log.Error("Invalid Request - PublishBatchV1")
+		return utils.CreateErrorResponseV1("InvalidParameterValue", false)
+	}
+
+	if requestBody.TopicArn == "" {
+		return utils.CreateErrorResponseV1("InvalidParameterValue", false)
+	}
+
+	topicArn := requestBody.TopicArn
+	arnSegments := strings.Split(topicArn, ":")
+	topicName := arnSegments[len(arnSegments)-1]
+	topic, ok := app.SyncTopics.Topics[topicName]
+	if !ok {
+		return utils.CreateErrorResponseV1("TopicNotFound", false)
+	}
+
+	seen := make(map[string]bool)
+
+	for _, entry := range requestBody.PublishBatchRequestEntries {
+		if entry.ID == "" {
+			// This is a required field, its absence likely indicates there are no further entries.
+			// It is unclear from the AWS docs if an error is returned if there are other fields
+			// present for PublishBatchRequestEntries.member.N where N is some integer in range [1,10].
+			break
```

Comment:
I suspect this is a required field for this `PublishBatchRequestEntry` entity, but it doesn't seem like it's REQUIRED in the request.  I think if they don't send it in, let's make one.  A new `uuid` should do it I think?